edx_django_utils/monitoring/internal/middleware.py

```diff
@@ -396,4 +399,19 @@ def get_log_message_and_monitor_cookies(self, request):
         else:
             log_prefix = f"Sampled small (< {logging_threshold}) cookie header."
         log_message = f"{log_prefix} BEGIN-COOKIE-SIZES(total={cookie_header_size}) {sizes} END-COOKIE-SIZES"
+
+        # If we have a large (or otherwise sampled) cookie header, and
+        # there's indication of corruption, just log all the headers
+        # for later diagnosis.
+        if corrupt_key_count:
```

Comment:
Hmm... yeah, probably worth casting a broader net, here.

Reply:
This might be a better query:

```
SELECT average(cookies.header.corrupt_count - cookies.header.corrupt_key_count) From Transaction WHERE cookies.header.corrupt_count IS NOT NULL and cookies.header.corrupt_key_count IS NOT NULL LIMIT MAX SINCE 4 weeks ago
```

It returns about 3.9, meaning that whenever "Cookie: " shows up at all, there are usually multiple additional corrupt values beyond what's in the keys. So yeah, I should use the overall count.

Reply:
This should be good enough to learn more about the issue. I will note that it may be possible that `Cookie: ` is even landing in other headers, but we'll know more when we can see some of them.